src/registry/editgrid/empty.ts

```diff
@@ -1,13 +1,53 @@
 import type {EditGridComponentSchema} from '@open-formulieren/types';
 
-import type {IsEmpty} from '@/registry/types';
+import type {GetRegistryEntry, IsEmpty} from '@/registry/types';
 import type {JSONObject} from '@/types';
 
-const isEmpty: IsEmpty<EditGridComponentSchema, JSONObject[]> = (_componentDefinition, value) => {
-  // Similar to Formio generic isEmpty https://github.com/formio/formio.js/blob/29939fc9d66f2b95527c90d3cf7729570c3d3010/src/components/_classes/component/Component.js#L3757
-  // The value of an editgrid component must be an array with 1 or more entries.
-  const hasValue = Array.isArray(value) && value.length >= 1;
-  return !hasValue;
+/**
+ * A function that validates the editgrid children components and returns whether they
+ * are considered as empty or not.
+ *
+ * Each (child) component is validated according to its `isEmpty` implemented function.
+ *
+ * @param parentComponentDefinition The parent (editgrid) component definition.
+ * @param componentValue The value of the editgrid component.
+ *
+ * @return The function returns whether the (child) component is empty (true) or not (false).
+ */
+export function isEmptyChildComponent(
+  parentComponentDefinition: EditGridComponentSchema,
+  componentValue: JSONObject,
+  getRegistryEntry: GetRegistryEntry
+): boolean | undefined {
+  const childrenKeys = Object.keys(componentValue);
+  if (childrenKeys.length === 0) return;
+
+  for (const childrenKey of childrenKeys) {
+    const value = componentValue[childrenKey];
+    const childDefinition = parentComponentDefinition.components.find(c => c.key === childrenKey);
+
+    if (!childDefinition) continue;
+
+    const registry = getRegistryEntry(childDefinition);
+    if (!registry?.isEmpty) continue;
+
+    const result = registry.isEmpty(childDefinition, value, getRegistryEntry);
```

Comment:
```suggestion
    const isChildEmpty = registry.isEmpty(childDefinition, value, getRegistryEntry);
```
Its a small detail, but I think its best to keep this function as clear as possible. (Just looking at myself, keeping track of the double negatives is already difficult enough)